KeywordSearch/src/org/sleuthkit/autopsy/keywordsearch/SolrSearchService.java

```diff
@@ -379,4 +381,33 @@ public void closeCaseResources(CaseContext context) throws AutopsyServiceExcepti
             throw new AutopsyServiceException(String.format("Failed to close core for %s", context.getCase().getCaseDirectory()), ex);
         }
     }
+
+    /**
+     * Deletes the solr core for a case. Finds the core name based on the caseDirectory if specified,
+     * otherwise will try to delete based on the textIndexName.
+     *
+     * @param textIndexName - the name of the core if the core is a solr 4 core
+     * @param caseDirectory - the case directory, used for getting the solr 6
+     *                      core name
+     *
+     * @throws KeywordSearchServiceException
+     */
+    @Override
+    public void deleteCore(String textIndexName, String caseDirectory) throws KeywordSearchServiceException {
+        String coreName = textIndexName;
+        if (!caseDirectory.isEmpty()) {
+            IndexMetadata indexMetadata;
+            try {
+                indexMetadata = new IndexMetadata(caseDirectory);
+                //When I was debugging and there were multiple cores they had the same name 
+                //so the first one is as good as any.
+                if (!indexMetadata.getIndexes().isEmpty()){
+                    coreName = indexMetadata.getIndexes().get(0).getIndexName();
```

Comment:
My previous solution which I decided was too convoluted that in Case before deleteTextIndex(metadata.getTextIndexName()) was called I checked if the arguement was equal to "" if it was I instead called searchservice.deleteTextIndex(metadata.getCaseDirectory()) instead and then inside deleteTextIndex I figured out if it was a path or just a core name if it was a path I got the IndexMetadata and checked to see if it had contents.   I however felt it was an inferior solution.